test/integ-new/main_new_test.go

```diff
@@ -212,47 +222,110 @@ func TestUIServiceHealthCheck(t *testing.T) {
 	}
 }
 
-func TestCertRotatorService(t *testing.T) {
+func TestCertRotatorHealthCheck(t *testing.T) {
 	t.Parallel()
 
-	cases := []struct {
-		name           string
-		path           string
-		wantStatusCode int
-	}{
-		{
-			name:           "cert_rotator_health_check",
-			path:           "/health",
-			wantStatusCode: http.StatusOK,
-		},
-		{
-			name:           "cert_rotator_key_rotation",
-			path:           "/",
-			wantStatusCode: http.StatusOK,
-		},
+	healthCheckPath := "/health"
+	wantStatusCode := http.StatusOK
+
+	ctx := context.Background()
+
+	uri := cfg.CertRotatorServiceAddr + healthCheckPath
+	resp := testSendHTTPReq(ctx, t, uri, cfg.CertRotatorServiceIDToken)
+
+	defer resp.Body.Close()
+
+	if got, want := resp.StatusCode, wantStatusCode; got != want {
+		b, err := io.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatal(err)
+		}
+		t.Errorf("Got unexpected status code, got=%d want=%d, response=%s", got, want, string(b))
 	}
+}
 
-	for _, tc := range cases {
-		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			t.Parallel()
+// Subtests must be run in sequence, and they have waits in between.
+// Therefore, they cannot be parallelized, and aren't a good fit for table testing.
+//
+//nolint:paralleltest
+func TestCertRotatorKeyRotation(t *testing.T) {
+	t.Parallel()
+	testutil.SkipIfNotIntegration(t)
```

Comment:
This is not required, as we did the check in TestMain

Reply:
Removed.